crates/sui-graphql-rpc/src/test_infra/cluster.rs

```diff
@@ -190,3 +189,73 @@ pub async fn start_test_indexer(
     });
     (store, handle)
 }
+
+pub async fn simulator_commands_test_impl() {
```

Comment:
This test impl shows how we can create custom chain state, and test against it

Reply:
Just out of curiosity as I am trying to make a mental model here, is it intended that for each test we'd add some custom chain state, or could we have 1-2-3 predefined custom chain states that are reused throughout tests?

Reply:
Both are doable. For the first case, we just extract out the final chain state and reuse it wherever we want in tests as seen here `simulator_commands_test_impl `.
For the second, case, we can rebuild state each time.

Reply:
Alright, because I would find it a bit cumbersome to always have to write these move files to create the chain state.